pkg/endpoint/endpoint.go

```diff
@@ -1797,3 +1802,71 @@ func (e *Endpoint) identityLabelsChanged(owner Owner, myChangeRev int) error {
 
 	return nil
 }
+
+// setPolicyRevision sets the policy wantedRev with the given revision.
+func (e *Endpoint) setPolicyRevision(rev uint64) {
+	e.policyRevision = rev
+	k := 0
+	for i, w := range e.policyRevisionSignals {
+		select {
+		case <-w.ctx.Done():
+			close(w.ch)
+		default:
+			if rev >= w.wantedRev {
+				close(w.ch)
+			} else {
+				// Make the closed policyRevisionSignals to go
+				// to the end of the slice so that we can keep them
+				// up afterwards
+				if i != k {
+					e.policyRevisionSignals[k] = w
+				}
+				k++
+			}
+		}
+	}
+	e.policyRevisionSignals = e.policyRevisionSignals[:k]
+}
+
+// cleanPolicySignals closes and removes all policy revision signals.
+func (e *Endpoint) cleanPolicySignals() {
+	for _, w := range e.policyRevisionSignals {
+		close(w.ch)
+	}
+	e.policyRevisionSignals = nil
```

Comment:
We seem to be trying to reuse `policyRevisionSignals` in `setPolicyRevision` so maybe do `e.policyRevisionSignals = e.policyRevisionSignals[:0]` ?